File: app/frontend/src/api/ai.ts
import { apiClient } from './client';

export interface AIStatus {
  enabled: boolean;
  configured: boolean;
  provider: string;
  model: string;
  mode: 'analysis_only';
}

export interface AIAnalysisResponse {
  analysis: string;
  provider: string;
  model: string;
  mode: 'analysis_only';
}

export async function getAIStatus(): Promise<AIStatus> {
  return apiClient.get<AIStatus>('/ai/status');
}

export async function requestAIAnalysis(context: Record<string, unknown>, question?: string): Promise<AIAnalysisResponse> {
  return apiClient.post<AIAnalysisResponse>('/ai/analyze', { context, question });
}

File: app/backend/app/models/ai.py
from typing import Any

from pydantic import BaseModel, Field


class AIAnalysisRequest(BaseModel):
    context: dict[str, Any] = Field(default_factory=dict)
    question: str | None = Field(default=None, max_length=1000)


class AIAnalysisResponse(BaseModel):
    analysis: str
    provider: str
    model: str
    mode: str = "analysis_only"
